File: testers/avro/scala-json/generated-and-checked-in/com/example/events/LinkedListNode.scala
package com.example.events

import com.fasterxml.jackson.annotation.JsonProperty

/** A recursive linked list for testing recursive type support */
case class LinkedListNode(
  /** The value stored in this node */
  @JsonProperty("value") value: Int,
  /** Optional next node in the list */
  @JsonProperty("next") next: Option[LinkedListNode]
)

File: testers/avro/java-spring/generated-and-checked-in/com/example/events/LinkedListNode.java
package com.example.events;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Optional;

/** A recursive linked list for testing recursive type support */
public record LinkedListNode(
    /** The value stored in this node */
    @JsonProperty("value") Integer value,
    /** Optional next node in the list */
    @JsonProperty("next") Optional<LinkedListNode> next) {
  /** The value stored in this node */
  public LinkedListNode withValue(Integer value) {
    return new LinkedListNode(value, next);
  }

  /** Optional next node in the list */
  public LinkedListNode withNext(Optional<LinkedListNode> next) {
    return new LinkedListNode(value, next);
  }
}
